File: Sistema_de_usuarios/ServicioDeLogueo.py
from RegistroDeCuenta import RegistroDeCuenta

class ServicioDeLogueo:
    def __init__(self, registro_cuentas):
        self.registro = registro_cuentas

    def autenticar(self, usuario: str, contraseña: str) -> bool:
        return self.registro.existe_usuario(usuario) and \
               self.registro.verificar_credenciales(usuario, contraseña)

File: Java/src/ejer/ServicioDeLogueo.java
package ejer;

public class ServicioDeLogueo {
	private RegistroDeCuenta registro;

    public ServicioDeLogueo(RegistroDeCuenta registroCuentas) {
        this.registro = registroCuentas;
    }

    public boolean autenticar(String usuario, String contraseña) {
        return registro.existeUsuario(usuario) &&
               registro.verificarCredenciales(usuario, contraseña);
    }
}
